### PR TITLE
Set the default minimum value of the QuantityBox to 0.

### PR DESCRIPTION
**Pull request by shaobinspire**

### CODE REVIEW

File: Applications/Spire/Source/Ui/QuantityBox.cpp [hunk 1]
```diff
@@ -8,6 +9,10 @@ namespace {
   struct QuantityToDecimalModel : ToDecimalModel<Quantity> {
     using ToDecimalModel<Quantity>::ToDecimalModel;
 
+    optional<Decimal> get_minimum() const override {
```
Comment (kranar) [inline Applications/Spire/Source/Ui/QuantityBox.cpp:12]: This is too narrow of a change. A ```QuantityBox``` that needs a non-negative value (aka unsigned) needs to use an unsigned model.

Comment (shaobinspire) [reply Applications/Spire/Source/Ui/QuantityBox.cpp:12]: I am little bit confused. Do you mean replace the Nexus::Quantity with unsigned int? Just like IntegerBox, the only difference is that the template parameter of the DecimalBoxAdaptor is unsigned int.

Comment (kranar) [reply Applications/Spire/Source/Ui/QuantityBox.cpp:12]: Any restrictions on QuantityBox's value shall be implemented in the model that is passed into the QuantityBox's constructor. A QuantityBox by itself does not restrict values, only display values, it is a view class. The data is part of the model and any properties of that data shall be implemented in the model.

**Comment by kranar** [comment]
This is way off and once again, making changes to the API without proposing them is taking a HUGE risk and will almost certainly result in wasted time and having those changes get rejected.

**Pull request closed without merging**